### PR TITLE
feat: GraphQL fragments

### LINKED ISSUE #3047: Add support for GraphQL fragments

### PR DESCRIPTION
**Pull request by nasdf**
## Relevant issue(s)

Resolves #3047

## Description

This PR enables support for GraphQL fragments.

## Tasks

- [x] I made sure the code is well commented, particularly hard-to-understand areas.
- [x] I made sure the repository-held documentation is changed accordingly.
- [x] I made sure the pull request title adheres to the conventional commit style (the subset used in the project can be found in [tools/configs/chglog/config.yml](tools/configs/chglog/config.yml)).
- [x] I made sure to discuss its limitations such as threats to validity, vulnerability to mistake and misuse, robustness to invalidation of assumptions, resource requirements, ...

## How has this been tested?

Added integration tests

Specify the platform(s) on which this was tested:
- MacOS

### CODE REVIEW

**Comment by jsimnz** [comment]
Reviewed but have a question. Should we also be supporting inline fragments? https://spec.graphql.org/October2021/#sec-Inline-Fragments

Fragments also support variables, if we want to support that too

**Comment by nasdf** [comment]
> Reviewed but have a question. Should we also be supporting inline fragments? https://spec.graphql.org/October2021/#sec-Inline-Fragments
> 
> Fragments also support variables, if we want to support that too

Nice catch! We should be able to support inline fragments as well as variables. I'll add some tests to verify.

File: tests/integration/query/simple/with_fragments_test.go [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+package simple
+
+import (
+	"testing"
+
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+)
+
+func TestQuerySimpleWithFragmentsSucceeds(t *testing.T) {
```
Comment (fredcarle) [inline tests/integration/query/simple/with_fragments_test.go:19]: suggestion: I'm not sure if you had changed them last time it was suggested but `TestQuerySimple_WithFragments_Succeeds` would read a little better and would be easier to go straight to the most descriptive part (`WithFragment`) of the test name.

Comment (nasdf) [reply tests/integration/query/simple/with_fragments_test.go]: done

**Review by fredcarle — APPROVED** [review]
LGTM! I'm surprised how simple supporting fragments turned out to be.

**Pull request merged**